src/dependency.py

```
@@ -18,20 +18,29 @@ def prompt_get_service_dependencies_and_dependents(service_id: str) -> str:
+        - 'security': A dependency on a service that handles authentication, authorisation, or request routing.
+        - 'performance': A dependency on a service that exists to optimise speed or reduce load.
+        - 'async': A dependency where the interaction is fire-and-forget or event-driven.
+        - 'config': A dependency on a service that provides configuration, feature flags, or secrets at runtime.
     """
 
 
 
 
 @dependency_mcp.tool()
-def create_dependency(service_id: str, dependency_id: str, version: str = None):
+def create_dependency(service_id: str, dependency_id: str, version: str = None, interaction_type: str = None):
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)